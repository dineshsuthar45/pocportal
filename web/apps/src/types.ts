import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
