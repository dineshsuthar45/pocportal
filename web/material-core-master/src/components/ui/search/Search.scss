.search-root {
	height: 18px;

	.MuiTextField-root {
		.MuiFilledInput-root:not(.Mui-disabled) {
			border-radius: 4px;

			&:before {
				border-bottom: none;
			}

			&:after {
				transition: none;
				transform: none;
				border-bottom: none;
			}

			&:hover:after {
				border-radius: 4px;
				height: 100%;
			}
		}

		.MuiFilledInput-root.Mui-disabled {
			border-radius: 4px;

			&::before {
				border-bottom-style: none;
			}

			&:after {
				transition: none;
				transform: none;
				border-bottom: none;
			}

			input {
				background-color: transparent;
			}
		}

		.MuiOutlinedInput-root {
			&.Mui-focused {
				.MuiOutlinedInput-notchedOutline {
					border-width: 1px;
				}
			}

			&:hover .MuiOutlinedInput-notchedOutline {
				border-width: 1px;
			}
		}

		.MuiInput-root {
			&:before {
				border-bottom: none;
			}

			&:after {
				transition: none;
				transform: none;
			}
		}
	}

	.adornment {
		cursor: pointer;
		&.search {
			padding-right: 5px;
		}

		&.reset {
			padding-left: 5px;

			&.hide {
				visibility: hidden;
			}
		}
	}

	.Mui-disabled {
		pointer-events: none;
	}

	input {
		padding: 0px;

		&::placeholder {
			color: var(--m-colour-greyScale-400);
			opacity: 1;
		}
	}
}
