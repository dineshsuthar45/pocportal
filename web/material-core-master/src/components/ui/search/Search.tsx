import { useTheme } from "@mui/material";
import classNames from "classnames";
import debounce from "lodash.debounce";
import React, { KeyboardEvent, useEffect, useRef, useState } from "react";
import { v4 as uuidv4 } from "uuid";
import { MColours } from "../../../theme/colours";
import { MIconSearchO, MIconX } from "../icons";
import { MTextfield } from "../textfield";
import "./Search.dark.scss";
import "./Search.light.scss";
import "./Search.scss";
import { SearchProps } from "./Search.types";

/**
 * Search component
 */
export const MSearch = React.forwardRef(
  (
    {
      debounceOnSearch = 250,
      disabled = false,
      id = uuidv4(),
      placeholder = "Search",
      variant = "filled",
      onChange,
      onReset,
      onSearch,
      ...props
    }: SearchProps,
    ref: React.ForwardedRef<HTMLDivElement>
  ) => {
    const debouncedSearch = useRef(
      debounce(onSearch, debounceOnSearch)
    ).current;

    const theme = useTheme();

    const inputSearch = useRef<HTMLInputElement | null>(null);

    const [initializationDone, setInitializationDone] = useState(false);
    const [showReset, setShowReset] = useState((props.value ?? "").length > 0);
    const [isFocused, setIsFocused] = useState(false);
    const [searchIcon, setSearchIcon] = useState(
      <MIconSearchO
        className="adornment search"
        height={18}
        width={18}
        fill={MColours.greyScale[800]}
      />
    );

    const [resetIcon, setResetIcon] = useState(
      <MIconX
        className="adornment reset"
        height={18}
        width={18}
        fill={MColours.greyScale[800]}
      />
    );

    const handleReset = (event: React.MouseEvent<SVGSVGElement>) => {
      onReset(event);
      inputSearch.current?.focus();
    };

    const handleSearch = () => {
      inputSearch.current?.focus();
    };

    const handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
      if (event.key === "Enter") {
        debouncedSearch.cancel();
        onSearch(props.value!);
      }
    };

    useEffect(() => {
      if (initializationDone) {
        debouncedSearch.cancel();
        debouncedSearch(props.value!);
        setShowReset((props.value ?? "").length > 0);
      }
    }, [props.value]);

    useEffect(() => {
      return () => {
        debouncedSearch.cancel();
      };
    }, [debouncedSearch]);

    useEffect(() => {
      const defaultColour = MColours.greyScale[400];
      const fillColor =
        theme.palette.mode === "light"
          ? isFocused
            ? "black"
            : defaultColour
          : isFocused
          ? "white"
          : defaultColour;

      setSearchIcon(
        <MIconSearchO
          className="adornment search"
          height={18}
          width={18}
          fill={fillColor}
          onClick={handleSearch}
        />
      );
      setResetIcon(
        <MIconX
          className={classNames("adornment reset", { hide: !showReset })}
          height={18}
          width={18}
          fill={fillColor}
          onClick={handleReset}
        />
      );
    }, [theme.palette.mode, isFocused, showReset]);

    useEffect(() => {
      setInitializationDone(true);
    }, []);

    return (
      <div className="search-root" ref={ref}>
        <MTextfield
          id={id}
          disabled={disabled}
          placeholder={placeholder}
          variant={variant}
          inputRef={inputSearch}
          InputProps={{
            onKeyDown: handleKeyDown,
            onFocus: () => setIsFocused(true),
            onBlur: () => setIsFocused(false),
            startAdornment: searchIcon,
            endAdornment: resetIcon,
          }}
          onChange={(e) => {
            setShowReset((e.target.value ?? "").length > 0);
            onChange!(e);
          }}
          {...props}
        />
      </div>
    );
  }
);

MSearch.defaultProps = {
  debounceOnSearch: 250,
  disabled: false,
  id: uuidv4(),
  placeholder: "Search",
  variant: "filled",
};
