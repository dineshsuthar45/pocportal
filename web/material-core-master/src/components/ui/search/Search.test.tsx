import { render } from "@testing-library/react";
import React from "react";
import { MSearch } from "./Search";
import { SearchProps } from "./Search.types";

describe("Search", () => {
  const renderComponent = () => render(<MSearch {...props} />);
  let props: SearchProps;

  beforeEach(() => {
    props = {
      id: "search-input",
      onChange: () => {},
      onReset: () => {},
      onSearch: () => {},
    };
  });

  it("should render correctly", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
