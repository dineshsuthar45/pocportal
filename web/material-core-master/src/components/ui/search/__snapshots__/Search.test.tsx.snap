// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly 1`] = `
<div>
  <div
    class="search-root"
  >
    <div
      class="MuiFormControl-root MuiTextField-root textfield-root light css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1kqdz9i-MuiInputBase-root-MuiFilledInput-root"
      >
        <svg
          class="adornment search"
          data-name="Layer 1"
          fill="#879BA6"
          height="18"
          viewBox="0 0 10 10"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.7 9.3 6.8 6.4A3.583 3.583 0 0 0 7.7 4 3.691 3.691 0 0 0 4 .3a3.7 3.7 0 1 0-.1 7.4 3.583 3.583 0 0 0 2.4-.9l2.9 2.9c.1.1.09.098.202.124a.292.292 0 0 0 .314-.12A.265.265 0 0 0 9.7 9.3ZM7 4a3.097 3.097 0 0 1-3 3.1A3.183 3.183 0 0 1 .9 4 3.116 3.116 0 0 1 4 .9 3.031 3.031 0 0 1 7 4Z"
          />
        </svg>
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-1o2n4nu-MuiInputBase-input-MuiFilledInput-input"
          id="search-input"
          placeholder="Search"
          type="text"
          value=""
        />
        <svg
          class="adornment reset hide"
          data-name="Layer 1"
          fill="#879BA6"
          height="18"
          viewBox="0 0 10 10"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.79 4.979 9.407 1.36a.563.563 0 1 0-.796-.796L4.993 4.182 1.375.565a.563.563 0 0 0-.796.796L4.197 4.98.579 8.596a.563.563 0 0 0 .796.797l3.618-3.618 3.618 3.618a.563.563 0 0 0 .796-.797Z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
